lib/absinthe/phase/document/validation/fields_on_correct_type.ex

```diff
@@ -193,7 +198,11 @@ defmodule Absinthe.Phase.Document.Validation.FieldsOnCorrectType do
   end
 
   defp suggested_field_names(external_field_name, %{fields: _} = type, blueprint) do
-    internal_field_name = blueprint.adapter.to_internal_name(external_field_name, :field)
+    internal_field_name =
+      case blueprint.adapter.to_internal_name(external_field_name, :field) do
+        nil -> external_field_name
```

Comment:
Where do these `suggested_*_names` changes come into play?

Reply:
The strict adapter returns `nil` if there a malformed name is seen.  Things down the line that depend on this value to be a binary break if `nil` is set to `internal_field_name` here.

The default adapter smooths over whatever mess `external_field_name` comes in looking like and it maps to a nice version of the internal name (so `FOO_BAR` becomes `foo_bar`).

In the strict adapter case, it makes more sense to return the original input (`FOO_BAR`) so it won't match correctly down the line and suggestions can be sent back to the client.

Reply:
sounds good